Messageモデルのテスト

avaklee opened the pull request:
# What
Rspec・factory girlを利用し、メッセージの保存できる場合とできない場合のテスト

#Why
バグがないことを確かめるため

avaklee committed:
Messageモデルのテスト

# What
Rspec・factory girlを利用し、メッセージの保存できる場合とできない場合のテスト

#Why
バグがないことを確かめるため

seoky0 commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,57 @@
+describe Message do
+  describe '#create' do
+    it "is valid with a message" do
+      message = build(:message, image: nil)
```
itの中身が全体的にインデントが深いので一つ浅くしてください

avaklee commented:
<img width="1440" alt="2017-06-09 20 32 28" src="https://user-images.githubusercontent.com/29111223/26973742-d3d5c15c-4d52-11e7-9077-40de59b3bd68.png">

seoky0 commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,57 @@
+  end
+end
+
+describe Message do
```
すでに定義しているので、再度定義する必要はありません。
最初に定義した

```
describe Message do 
  describe '#create' do
```

の中に複数のテストを書いていただければ大丈夫です。

seoky0 commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,57 @@
+
+describe Message do
+  describe '#create' do
+    it "is invalid without neither message nor image" do
```
保存できてなかったときにエラーメッセージが含まれているかのテストもしましょう

avaklee committed:
Messageモデルのテストの修正

# What

インデント、再度定義していた部分の簡略化、保存されなかった時のエラーメッセージを含むテストの追加
#Why

みやすくするため、
テストの内容でエラーメッセージまで含む

avaklee commented:
<img width="1440" alt="2017-06-09 20 49 00" src="https://user-images.githubusercontent.com/29111223/26974177-254da7f0-4d55-11e7-834c-44a9a2df3ebe.png">

yusukeyamane commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,39 @@
+      expect(message).to be_valid
+    end
+
+    it "is invalid without neither message nor image" do
```
基本的に１つのitでexpectはひとつになるようにしましょう

yusukeyamane commented on spec/models/message_spec.rb:
```diff
@@ -0,0 +1,39 @@
+
+    it "is invalid without neither message nor image" do
+      message = create(:message)
+      message.image = nil
```
message.image = nil
message.body = nil
この二行は一行でかけます。

avaklee committed:
Message Test

# What
二行を一行に、またテストを一つに変更
#Why
みやすく、管理しやすくするため

ghost commented:
コメントにはリアクションか、返信をするようにしましょう。
理由は、私たちがレビューを見る際に、こちらがコメントを見ているか、そして修正できているかを確認しているためです

hiroki5123 commented on spec/factories/messages.rb:
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+
+  factory :message do
+    body "Hi, world"
```
fakerを入れているので、使うようにしましょう。
bodyに関しては、Faker::Lorem.sentenceでセンテンスをつくることができますので、一度ご確認ください。

ghost commented:
修正後、もう一度テスト結果をこちらに貼り付けていただくようお願い致します。

avaklee committed:
Use of Faker

# What
Use Faker for Body

#Why
To test

avaklee commented:
<img width="1440" alt="2017-06-12 11 28 23" src="https://user-images.githubusercontent.com/29111223/27017279-4b205414-4f62-11e7-9352-59f8c97ac503.png">

ghost commented:
リアクションありがとうございます！
これからこちらもリアクションかコメントをつけていただけるとありがたいです。
よろしくお願い致します。

![2017-06-12 12 42 29](https://user-images.githubusercontent.com/23256513/27018591-b25a748e-4f6c-11e7-850b-0505ff293b4a.png)

ghost commented:
![space](https://user-images.githubusercontent.com/23256513/27018673-1f08c20c-4f6d-11e7-9d51-8a0cc3c9c0c6.gif)
